docker/startup.sh

```diff
@@ -2,8 +2,13 @@
 # The purpose of this script is to set a log regex to forward error logs to stderr and other logs to
 # stdout. This allows us to easily configure alerts when any message goes to stderr.
 
-# The main command is run in the background so that it receives interrupt signals
+# Run both commands in the background, so that we can receive signals and forward it to the main process (for this we capture the pid).
 # The stream-split command deliberately ignores interrupts (otherwise we may lose logs on teardown). It terminates automatically
 # once the main command (stdin) terminates.
-("$@" &) | (trap '' INT; regex-stream-split '^\d+-\d+-\d+T\d+:\d+:\d+\.\d+Z\s+(TRACE|DEBUG|INFO|WARN)' '^\d+-\d+-\d+T\d+:\d+:\d+\.\d+Z\s+ERROR')
+( "$@" & echo $! > main_pid ) | (trap '' TERM INT; regex-stream-split '^\d+-\d+-\d+T\d+:\d+:\d+\.\d+Z\s+(TRACE|DEBUG|INFO|WARN)' '^\d+-\d+-\d+T\d+:\d+:\d+\.\d+Z\s+ERROR') &
 
+pid=$(cat main_pid)
+trap 'echo foo; kill -15 $pid' TERM
+trap 'echo bar; kill -2 $pid' INT
```

Comment:
Are foo and bar expected to be echoed when we terminate the process now? What ever happened to `kill -9`?

Reply:
This is outdated